### DISCUSSION TITLE
Unable to start AWS CloudWatch Agent with `airflow` user

### DISCUSSION THREAD

**Discussion by RajasGujarathi** [apache/airflow#40404] [Q&A] [answered]
Using the docker image `apache/airflow:slim-2.9.2-python3.12`

We want to use the AWS CloudWatch agent to be able to push logs from the webserver, scheduler & worker to AWS CloudWatch logs. We are running the AWS CloudWatch agent as a non-root user (`airflow` user),

`/opt/aws/amazon-cloudwatch-agent/bin/start-amazon-cloudwatch-agent &`

however, while starting the CloudWatch agent we get the following error:

``` bash
.
.
.
    mode = "OP"
    profile = "AmazonCloudWatchAgent"
    region = "eu-west-1"
    region_type = "ACJ"
    shared_credential_file = "/home/airflow/.aws/credentials"
2024/06/24 13:38:35 I! Config has been translated into YAML /opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.yaml 
2024/06/24 13:38:35 Reading json config file path: /opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.json ...
2024/06/24 13:38:35 I! Valid Json input schema.
2024/06/24 13:38:35 I! Detected runAsUser: airflow
2024/06/24 13:38:35 I! Changing ownership of [/opt/aws/amazon-cloudwatch-agent/logs /opt/aws/amazon-cloudwatch-agent/etc /opt/aws/amazon-cloudwatch-agent/var] to 50000:0
2024/06/24 13:38:35 E! Failed to set groups: operation not permitted
2024/06/24 13:38:35 E! failed switching to "airflow": operation not permitted
2024/06/24 13:38:35 E! Failed to ChangeUser: operation not permitted 
2024/06/24 13:38:35 E! Error when starting Agent, Error is operation not permitted 
```
<br>
Not sure what we are missing there🤞🤔 . Any help will be appreciated

---

> #### Note 
> For debugging, when we switched to `root` user we were able to start AWS CloudWatch agent and it worked as expected.

**Comment by potiuk** [comment]
I think it's a great question for CloudWatch support.

**Reply by RajasGujarathi** [reply]
Yes, have asked them as well 🤞

**Reply by RajasGujarathi** [reply, marked as answer]
For someone in the future, we had to run the following command to make this work.

> sudo setcap cap_setgid=pe /opt/aws/amazon-cloudwatch-agent/bin/start-amazon-cloudwatch-agent

Note: The command must be run with `sudo`, and then the `airflow` user can start the AWS Cloudwatch agent.

``` bash
.
.
sudo setcap cap_setgid=pe /opt/aws/amazon-cloudwatch-agent/bin/start-amazon-cloudwatch-agent
.
.
su airflow
.
.
/opt/aws/amazon-cloudwatch-agent/bin/start-amazon-cloudwatch-agent &
```

Thanks to AWS Support 🙏 for the suggestion & fix.
